src/Microsoft.Extensions.CodeGeneration.Core/DefaultCodeGeneratorAssemblyProvider.cs

```diff
@@ -54,8 +51,12 @@ public IEnumerable<Assembly> CandidateAssemblies
                     .SelectMany(_libraryManager.GetReferencingLibraries)
                     .Distinct()
                     .Where(IsCandidateLibrary);
-                
+#if NET451
+                // For Net 451 the exe is being invoked from the build folder so all dependencies are next to the exe.
+                return list.Select(lib => _assemblyLoadContext.LoadFromName(new AssemblyName(lib.Identity.Name)));
```

Comment:
@natemcmaster Scaffolding does try to host the user's Startup.cs to retrieve model metadata information. 
I just tried with a project that referenced Newtonsoft.Json 8.0.3, and it seems to work fine. 
It may be because scaffolding doesn't have a dependency on ProjectModel.Loader